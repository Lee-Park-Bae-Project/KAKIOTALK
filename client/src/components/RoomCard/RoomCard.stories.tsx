import React from 'react';
import RoomCard from 'components/RoomCard';
import faker from 'faker';
import styled from 'styled-components';

export default {
  title: 'Component/RoomCard',
  component: RoomCard,
};

export const Default = () => {
  const userList = [
    faker.helpers.userCard().name,
    faker.helpers.userCard().name,
    faker.helpers.userCard().name,
  ];
  const lastMessage = faker.lorem.paragraph();
  const lastModified = Date.now();
  const numOfNewMessages = Math.round(Math.random() * 100);
  const Wrapper = styled.div`
    width: 25rem;
  `;
  return (
    <Wrapper>
      <RoomCard
        userList={userList}
        lastMessage={lastMessage}
        lastModified={lastModified}
        numOfNewMessages={numOfNewMessages}
        />
    </Wrapper>
  );
};
