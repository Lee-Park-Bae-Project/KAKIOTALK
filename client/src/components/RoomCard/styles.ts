import styled, { css } from 'styled-components';
import Flex from 'commons/Flex';
import { color, weight } from 'styles/global';

export const grayColor = css`
  color: ${(props) => color.GRAY};
`;

export const Container = styled(Flex)`
  height: 5rem;
  margin: 0.5rem 0;
  min-width: 10rem;
  justify-content: space-between;
  &:hover{
    background: ${color.HOVER_GRAY};
  }
`;

export const RoomInfoWrapper = styled.div`
  display: flex;
  max-width: 80%;
  

`;

export const ImgWrapper = styled.div`
  margin: 1rem;
  width: auto;
`;

export const InfoWrapper = styled.div`
  flex-direction: column;
  align-items: flex-start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  min-width: 80%;
  width: 100%;
`;

export const Row = styled.div`
  justify-content: left;
  /* min-width: 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 80%; */
`;

export const UserListWrapper = styled.span`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  margin-right: 1rem;
  min-width: 0;
  display: block;
  font-weight: ${weight.STRONG};
`;

export const NumWrapper = styled.span`
  ${(props) => grayColor};
  /* display: inline-flex; */

  /* color: ${color.GRAY}; */
`;

export const LastMsgWrapper = styled.div`
  ${(props) => grayColor};
  margin-right: 1rem;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  /* width: 10rem; */
  height: 2.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const SubInfoWrapper = styled(Flex)`
  flex-direction: column;
  align-items: flex-end;
  flex:1
  span{
    ${(props) => grayColor};
  }  
`;

export const Time = styled.span`
  font-size: 0.7rem;
`;
