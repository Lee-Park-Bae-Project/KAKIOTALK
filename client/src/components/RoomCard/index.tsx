import React, { FC } from 'react';
import * as S from 'components/RoomCard/styles';
import Icon from 'Icon/Icon';
import Circle from 'atoms/Circle';
import { convertMillToMMDDYYYY } from 'common/utils';

interface Props{
  participantsName: string;
  lastMessage?: string;
  lastModified?: number;
  numOfNewMessages?: number;
  onClick?: () => void;
}
const ChatCard: FC<Props> = ({
  participantsName,
  lastMessage = 'this is last message',
  lastModified = Date.now(),
  numOfNewMessages = 99,
  onClick = undefined,
}) => (
    <S.Container onClick={onClick}>
      <S.RoomInfoWrapper>
        <S.ImgWrapper>
          <Icon icon="Person"/>
        </S.ImgWrapper>
        <S.InfoWrapper>
          <S.Row>
            <div>
              <S.UserListWrapper>
                {participantsName}
              </S.UserListWrapper>
            </div>
          </S.Row>
          <S.LastMsgWrapper>
            <span>{lastMessage}</span>
          </S.LastMsgWrapper>
        </S.InfoWrapper>
        <S.NumWrapper>
          {participantsName.length}
        </S.NumWrapper>
      </S.RoomInfoWrapper>

      <S.SubInfoWrapper>
        <S.Time>{convertMillToMMDDYYYY(lastModified)}</S.Time>
        <Circle num={numOfNewMessages}/>
      </S.SubInfoWrapper>
    </S.Container>
);

export default ChatCard;
