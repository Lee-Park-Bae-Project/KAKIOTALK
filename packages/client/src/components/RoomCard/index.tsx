import React, { FC } from 'react'
import * as S from 'components/RoomCard/styles'
import Icon from 'Icon/Icon'
import {
  convertToLL, convertToLT, getCurTimeDBFormat, getDayDiff,
} from 'common/utils'

interface Props {
  participantsName: string;
  numOfParticipants: number;
  lastMessage?: string;
  lastModified?: string;
  onClick?: () => void;
}
const RoomCard: FC<Props> = ({
  participantsName,
  numOfParticipants,
  lastMessage = 'this is last message',
  lastModified = getCurTimeDBFormat(),
  onClick = undefined,
}) => {
  const dayDiffFromNow = getDayDiff(lastModified)
  return (
  <S.Container onClick={onClick}>
    <S.RoomInfoWrapper>
      <S.ImgWrapper>
        <Icon icon="Person" />
      </S.ImgWrapper>
      <S.InfoWrapper>
        <S.Row>
            <S.UserListWrapper>{participantsName}</S.UserListWrapper>
            <S.NumWrapper>{numOfParticipants}</S.NumWrapper>
        </S.Row>
        <S.LastMsgWrapper>
          {lastMessage}
        </S.LastMsgWrapper>
      </S.InfoWrapper>
    </S.RoomInfoWrapper>
    <S.SubInfoWrapper>
      <S.Time>{ dayDiffFromNow ? convertToLT(lastModified) : convertToLL(lastModified)}</S.Time>
    </S.SubInfoWrapper>
  </S.Container>
  )
}

export default RoomCard
