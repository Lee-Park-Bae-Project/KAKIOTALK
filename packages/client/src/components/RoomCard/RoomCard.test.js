/* eslint-disable no-undef */
import React from 'react'
import { mount } from 'enzyme'
import RoomCard from 'components/RoomCard'

// faker.locale = 'ko';
describe('<RoomCard/>', () => {
  const participantsName = ['a', 's', 'd', 'c']
  const lastMessage = 'hihi'
  const lastModified = '2020-04-25 22:13:52'

  it('matches snapshots', () => {
    const wrapper = mount(
      <RoomCard
        participantsName={participantsName}
        lastMessage={lastMessage}
        lastModified={lastModified}
        numOfNewMessages={3}
    />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
