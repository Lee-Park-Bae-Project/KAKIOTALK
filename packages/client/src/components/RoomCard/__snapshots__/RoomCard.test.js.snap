// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomCard/> matches snapshots 1`] = `ReactWrapper {}`;
